goal_src/levels/common/joint-exploder.gc

```diff
@@ -59,53 +63,623 @@
-   (static-params              joint-exploder-static-params                         :offset-assert 188)
+  ((parent-override            (pointer process-drawable)             :offset         12)
+   (die-if-below-y             float                                  :offset-assert 176)
+   (die-if-beyond-xz-dist-sqrd float                                  :offset-assert 180)
+   (joints                     joint-exploder-joints                  :offset-assert 184)
+   (static-params              joint-exploder-static-params           :offset-assert 188)
    (anim                       art-joint-anim                         :offset-assert 192)
-   (scale-vector               vector                :inline :offset-assert 208)
-   (tuning                     joint-exploder-tuning :inline :offset-assert 224)
-   (lists                      joint-exploder-list               5      :inline :offset-assert 288)
+   (scale-vector               vector                       :inline   :offset-assert 208)
+   (tuning                     joint-exploder-tuning        :inline   :offset-assert 224)
+   (lists                      joint-exploder-list          5 :inline :offset-assert 288)
    )
+  :heap-base #x1a0
   :method-count-assert 29
   :size-assert         #x210
-  :heap-base           #x1a0
   :flag-assert         #x1d01a00210
   (:methods
-    (dummy-20 (_type_) none 20)
-    (dummy-21 (_type_) none 21)
-    (dummy-22 () none 22)
-    (dummy-23 () none 23)
-    (dummy-24 (_type_) none 24)
-    (dummy-25 () none 25)
-    (dummy-26 (_type_) none 26)
-    (dummy-27 (_type_ joint-exploder-list) none 27)
-    (dummy-28 (_type_) none 28)
+    (TODO-RENAME-20 (_type_ joint-exploder-list int) int 20)
+    (TODO-RENAME-21 (_type_ joint-exploder-list joint-exploder-joint) none 21)
+    (TODO-RENAME-22 (_type_ joint-exploder-list) symbol 22)
+    (TODO-RENAME-23 (_type_) symbol 23)
+    (TODO-RENAME-24 (_type_ joint-exploder-list int) int 24)
+    (TODO-RENAME-25 (_type_ joint-exploder-list) symbol 25)
+    (TODO-RENAME-26 (_type_ joint-exploder-list int) int 26)
+    (TODO-RENAME-27 (_type_ joint-exploder-list int) joint-exploder-list 27)
+    (TODO-RENAME-28 (_type_ joint-exploder-list) none 28)
+    )
+  )
+
+
+(defmethod asize-of joint-exploder-joints ((obj joint-exploder-joints))
+  (the-as int (+ (-> obj type size) (* 176 (-> obj num-joints))))
+  )
+
+(defmethod new joint-exploder-joints ((allocation symbol) (type-to-make type) (arg0 joint-exploder-static-params))
+  (let* ((gp-0 (-> arg0 joints length))
+         (v0-0 (object-new allocation type-to-make (the-as int (+ (-> type-to-make size) (* 176 gp-0)))))
+         )
+    (set! (-> v0-0 num-joints) gp-0)
+    v0-0
+    )
+  )
+
+(defun joint-exploder-joint-callback ((arg0 joint-exploder))
+  (let ((gp-0 (-> arg0 node-list)))
+    (let ((s4-0 (-> arg0 joints)))
+      (countdown (s3-0 (-> s4-0 num-joints))
+        (let* ((v1-3 (the-as joint-exploder-joint (+ (+ (* 176 s3-0) 12) (the-as int s4-0))))
```

Comment:
this should hopefully become `(-> s4-0 joint s3-0)` (and similar for all the `(* 176 ...)` stuff with casts